internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -557,6 +557,7 @@ func (c *enrollCmd) enrollWithBackoff(ctx context.Context, persistentConfig map[
 			retry = true
 		case err != nil:
 			c.log.Warnf("Enrollment failed: %s", err.Error())
+			retry = true
```

Comment:
In all cases it seems that `retry` is `true`. We should update the code to just remove the `retry` variable.